pkgs/tools/text/diffsitter/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, fetchFromGitHub
+, linkFarm
+, makeWrapper
+, rustPlatform
+, tree-sitter
+}:
+
+let
+  # based on https://github.com/NixOS/nixpkgs/blob/aa07b78b9606daf1145a37f6299c6066939df075/pkgs/development/tools/parsing/tree-sitter/default.nix#L85-L104
+  withPlugins = grammarFn:
+    let
+      grammars = grammarFn tree-sitter.builtGrammars;
+    in
+    linkFarm "grammars"
+      (map
+        (drv:
+          let
+            name = lib.strings.getName drv;
+          in
+          {
+            name =
+              "lib" +
+              (lib.strings.removeSuffix "-grammar" name)
+              + ".so";
+            path = "${drv}/parser";
+          }
+        )
+        grammars);
+
+  libPath = withPlugins (_: tree-sitter.allGrammars);
```

Comment:
diffsitter has [a list of supported languages](https://github.com/afnanenayet/diffsitter#summary), perhaps it is best to default to only the supported languages

i think it would be easier to just linkFarm from `tree-sitter.builtGrammars` or `tree-sitter-grammars` so we don't need stuff like `getName` or `removeSuffix`

Reply:
I don't think it really matters, the list of supported languages is just a list of tree-sitter grammars I vendored, but there's nothing language specific in the code after running the parser. All that matters is that you're linking a  compatible dynamic library.

Reply:
@afnanenayet thanks for the explanation, both none-officialy-supported files (lua and nix) happened to not have [file associations](https://github.com/afnanenayet/diffsitter/blob/v0.7.1/src/parse.rs#L27), but its easy to add file assocations in the config so i agree with the decision to default to all grammars

my point with using `tree-sitter.builtGrammars` or `tree-sitter-grammars` still stands though

Reply:
Yep totally valid. I have no objections there, thanks for flagging that I don't have file associations for Lua/Nix, I'll add them.

Reply:
> i think it would be easier to just linkFarm from `tree-sitter.builtGrammars` or `tree-sitter-grammars` so we don't need stuff like `getName` or `removeSuffix`

@figsoda: if it's good enough for tree-sitter (https://github.com/NixOS/nixpkgs/blob/aa07b78b9606daf1145a37f6299c6066939df075/pkgs/development/tools/parsing/tree-sitter/default.nix#L85-L104) could we just merge this and have another PR later to improve both packages?

This PR is getting old.